lib/ddtrace/contrib/aws/instrumentation.rb

```diff
@@ -0,0 +1,43 @@
+module Datadog
+  module Contrib
+    module Aws
+      # A Seahorse::Client::Plugin that enables instrumentation for all AWS services
+      class Instrumentation < Seahorse::Client::Plugin
+        def add_handlers(handlers, _)
+          handlers.add(Handler, step: :validate)
+        end
+      end
+
+      # Generates Spans for all interactions with AWS
+      class Handler < Seahorse::Client::Handler
+        def call(context)
+          pin = Datadog::Pin.get_from(::Aws)
+
+          return @handler.call(context) unless pin && pin.tracer
+
+          pin.tracer.trace(RESOURCE) do |span|
+            result = @handler.call(context)
+            annotate!(span, pin, ParsedContext.new(context))
+            result
+          end
+        end
+
+        private
+
+        def annotate!(span, pin, context)
+          span.service = pin.service
+          span.span_type = pin.app_type
+          span.name = context.safely(:resource)
```

Comment:
what kind of names do we have here? can we check that in our tests also so it's explicit what we're expecting?

Reply:
Basically I tried to mimmic the Python's client behavior here. The name will be something like `service.action`. There is a test case for that already: https://github.com/DataDog/dd-trace-rb/blob/54ed5b7212e506f9b8304ee0361d2052f3eaf01e/test/contrib/aws/parsed_context_test.rb#L12-L14

Reply:
Whoops, just realized we don't have a explicit test for the name! Will add that!

Reply:
👍